### PR TITLE
Refactored tokenizers

### PR DESCRIPTION
**Pull request by cjweir**
They can now do lower-case and n-gram contraction was separated into a separate transformer.

### CODE REVIEW

File: textmap/tests/test_tokenizers.py [hunk 1]
```diff
@@ -12,26 +13,39 @@
...
+from textmap.tranformers import MWETransformer
 
 
 def test_sklearn_tokenizer():
     for tokens_by in ["document", "sentence"]:
-        tokenizer = SKLearnTokenizer(tokenize_by=tokens_by).fit(test_text)
+        for lower in [True, False]:
+            tokenizer = SKLearnTokenizer(tokenize_by=tokens_by, lower_case=lower).fit(
+                test_text
+            )
```
Comment (lmcinnes) [inline textmap/tests/test_tokenizers.py:24]: Pytest has some handy options for this sort of thing. It isn't required, but you can do something like:

```python
@pytest.mark.parameterize("tokens_by", ["document", "sentence"])
@pytest.mark.parameterize("lower", [True, False])
def test_sklearn_tokenizer(tokens_by, lower):
    tokenizer = SKLearnTokenizer(tokenize_by=tokens_by, lower_case=lower).fit(
                test_text
            )
```

while this is more actual code, it makes the test reports a little nicer. Up to you whether you want to bother with such things for these cases.

File: textmap/tokenizers.py [hunk 2]
```diff
@@ -182,129 +131,93 @@ def fit(self, X, **fit_params):
...
-                self.tokenization_ = [word_tokenize(doc) for doc in X]
+            tokenize = lambda d: self.nlp.tokenize(d)
 
-        self.iteratively_contract_bigrams()
+        if self.tokenize_by == "sentence":
+            self.tokenization_ = flatten(
+                [[tokenize(sent) for sent in sent_tokenize(doc)] for doc in X]
+            )
+        else:
+            self.tokenization_ = [tokenize(doc) for doc in X]
```
Comment (lmcinnes) [inline textmap/tokenizers.py:145]: You may want to make this an
```suggestion
        elif self.tokenize_by == "document":
            self.tokenization_ = [tokenize(doc) for doc in X]
        else:
            raise ValueError(f"Unrecognized tokenize_by argument {self.tokenize_by}. Should be one of \"sentence\" or \"document\"")
```

File: textmap/tokenizers.py [hunk 3]
```diff
@@ -317,17 +230,13 @@ def fit(self, X, **fit_params):
...
-                for doc in X
-            ]
+            self.tokenization_ = flatten(
+                [[self.nlp(sent) for sent in sent_tokenize(doc)] for doc in X]
+            )
         else:
-            self.tokenization_ = [sk_word_tokenize(sk_preprocesser(doc)) for doc in X]
-
-        self.iteratively_contract_bigrams()
+            self.tokenization_ = [self.nlp(doc) for doc in X]
```
Comment (lmcinnes) [inline textmap/tokenizers.py:239]: As per the earlier comment; you may want to fail on unreconized strings... I guess this applies throughout.

File: textmap/tokenizers.py (hunk 2 above)
Comment (cjweir) [reply textmap/tokenizers.py:145]: Cool.  I can do that.  There were a few typo's that I'm fixing up as well.

File: textmap/tests/test_tokenizers.py (hunk 1 above)
Comment (cjweir) [reply textmap/tests/test_tokenizers.py:24]: I can.  I've got all afternoon.  I'm just fixing a few typo bugs that I just found.

File: textmap/tranformers.py [hunk 4]
```diff
@@ -390,3 +395,80 @@ def fit_transform(self, X, y=None, **fit_params):
         )
         self.mix_weights_ = weights
         return result
+
+
+class MWETransformer(BaseEstimator, TransformerMixin):
```
Comment (jc-healy) [inline textmap/tranformers.py:400]: I'd be tempted to change this the very verbose MultipleWordExpressionTransformer.    Most non-NLP folks aren't going to know what MWE stands for.

**Pull request merged**